### PR TITLE
Folders - Permanently Delete

### PR DESCRIPTION
**Pull request by Trudko**
Added unit test.
Added implementation.
Added documentation.

### CODE REVIEW

File: lib/managers/folders.js [hunk 1]
```diff
@@ -301,6 +301,21 @@ Folders.prototype.deleteMetadata = function(folderID, scope, template, callback)
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+/**
+ * Permanently deletes an folder that is in the trash. The item will no longer exist in Box. This action cannot be undone
+ *
+ * API Endpoint: '/folders/:folderID/trash/'
+ * Method: DELETE
+ *
+ * @param  {type} folderID Box ID of the folder being requested
```
Comment (mattwiller) [inline lib/managers/folders.js]: This should have type `{string}`

File: tests/lib/managers/folders-test.js [hunk 2]
```diff
@@ -276,4 +276,20 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('permanentlyDelete()', function() {
```
Comment (mattwiller) [inline tests/lib/managers/folders-test.js]: Should match the method name, `deletePermanently()`

**Comment by mattwiller** [comment]
Looks good, just a couple minor comments.  If you can fix those up, it should be ready to merge!

File: tests/lib/managers/folders-test.js [hunk 3]
```diff
@@ -276,4 +276,20 @@ describe('Folders', function() {
...
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').returns(done);
+			sandbox.stub(boxClientFake, 'del').withArgs('/folders/' + FOLDER_ID + '/trash').yieldsAsync();
+			folders.deletePermanently(FOLDER_ID, done);
+		});
+
+		it('should make DELETE call to remove folder permanently', function(done) {
+
+			sandbox.stub(boxClientFake, 'defaultResponseHandler').yields();
+			sandbox.mock(boxClientFake).expects('del').withArgs('/folders/' + FOLDER_ID + '/trash', null);
```
Comment (mattwiller) [inline tests/lib/managers/folders-test.js]: This stub/mock combination isn't quite right; `.yields()` means that `defaultResponseHandler()` will call its function argument immediately when called.  In this test, that argument is `done`, so the test will end immediately on that line.  This is incorrect; `defaultResponseHandler()` should instead return a function and `boxClient.del()` should yield asynchronously to it.  See your above test.

File: tests/lib/managers/folders-test.js [hunk 4]
```diff
@@ -276,4 +276,20 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('deletePermanently()', function() {
+
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').returns(done);
```
Comment (mattwiller) [inline tests/lib/managers/folders-test.js]: I would change this to either `.withArgs(done).returns(done)` or simply `.returnsArg(0)` to capture the expected behavior around `defaultResponseHandler()`, namely that is is passed a callback and produces a new callback that should be called once the operation is complete.

File: lib/managers/folders.js [hunk 5]
```diff
@@ -301,6 +301,21 @@ Folders.prototype.deleteMetadata = function(folderID, scope, template, callback)
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+/**
+ * Permanently deletes an folder that is in the trash. The item will no longer exist in Box. This action cannot be undone
+ *
+ * API Endpoint: '/folders/:folderID/trash/'
```
Comment (mattwiller) [inline lib/managers/folders.js]: Super minor nitpick, but would you mind removing the trailing slash after `/trash` here?

File: tests/lib/managers/folders-test.js (hunk 4 above)
Comment (Trudko) [inline tests/lib/managers/folders-test.js]: That is what I wanted to ask , many tests are using withArgs('/folders/1234/metadata/global/properties... Why is that? Wouldn't it make sense only if you want to test with different arguments and test different possible responses?

File: lib/managers/folders.js (hunk 5 above)
Comment (Trudko) [inline lib/managers/folders.js]: Sure , don't worry about nitpicks let's do it right. Need to be bit more careful about this.

File: tests/lib/managers/folders-test.js (hunk 4 above)
Comment (mattwiller) [inline tests/lib/managers/folders-test.js]: We use `.withArgs()` to assert that the mocked function has the correct arguments passed to it.

Comment (Trudko) [inline tests/lib/managers/folders-test.js]: @mattwiller how would the result test look like? All other tests of this kind are in this format, but I think there might be a room for improvement. We might discuss it closer over hangouts.

Comment (Trudko) [inline tests/lib/managers/folders-test.js]: I've used .withArgs(done).returns(done) just to be more explicit.

**Comment by mattwiller** [comment]
Looks good, you'll just need to resolve the merge conflicts.

**Pull request merged**